src/Engines/Database.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace WTSA1\Engines;
+
+// General singleton class.
+class Database {
+    // Hold the class instance.
+    private static $instance = null;
+    private $db;
+    
+    // The constructor is private
+    // to prevent initiation with outer code.
+    private function __construct($testMode = false)
+    {
+        define('DBHost', $testMode ? 'test_mysql' : 'mysql');
+        define('DBPort', getenv('MYSQL_PORT'));
+        define('DBName', getenv('MYSQL_DATABASE'));
+        define('DBUser', getenv('MYSQL_USER'));
+        define('DBPassword', getenv('MYSQL_PASSWORD'));
+
+        $this->db = new \Db(DBHost, DBPort, DBName, DBUser, DBPassword);
```

Comment:
Maybe we should `exit()` if any of these is not present. This would make it easier to debug the deployment if the environment is not correct. - 🤔 mhh

Reply:
Good idea, gonna add this condition now.

Reply:
https://github.com/IBZTI2018/TI3-WT-SA/commit/4af9cdfd155358d5bf85215f3f4946adc1bd6460